include/ttmlir/Dialect/TTKernel/IR/TTKernelOps.td

```diff
@@ -662,6 +675,35 @@ def TTKernel_GetInterleavedAddrGenFastOp : TTKernel_Op<"get_interleaved_addr_gen
     let results = (outs TTKernel_InterleavedAddrGenFast:$result);
 }
 
+def TTKernel_MyXOp : TTKernel_Op<"my_x"> {
+    let summary = "MyX";
+    let description = [{
+      Lowers to the tt-metal supported MY_X macro. This represents the virtual X coordinate of the current core.
+    }];
+
+    let arguments = (ins Optional<I8>:$noc);
+    let results = (outs Index:$x);
+}
+
+def TTKernel_MyYOp : TTKernel_Op<"my_y"> {
+    let summary = "MyY";
+    let description = [{
+      Lowers to the tt-metal supported MY_Y macro. This represents the virtual Y coordinate of the current core.
+    }];
+
+    let arguments = (ins Optional<I8>:$noc);
+    let results = (outs Index:$y);
+}
+
+def TTKernel_GetCBOp : TTKernel_Op<"get_cb"> {
+  let summary = "Get CB";
+  let description = [{
+     Returns a CB type.
+    }];
+
+  let results = (outs TTKernel_CB:$cb);
```

Comment:
Is this op equivalent to `create_buffer` but it returns a `TTKernel_CB` instead of a `AnyNon0RankedMemref`?